spec.md

```diff
@@ -2315,6 +2348,8 @@ The CO MUST implement the specified error recovery behavior when it encounters t
 | Volume published but is incompatible | 6 ALREADY_EXISTS | Indicates that a volume corresponding to the specified `volume_id` has already been published at the specified `target_path` but is incompatible with the specified `volume_capability` or `readonly` flag. | Caller MUST fix the arguments before retrying. |
 | Exceeds capabilities | 9 FAILED_PRECONDITION | Indicates that the CO has exceeded the volume's capabilities because the volume does not have MULTI_NODE capability. | Caller MAY choose to call `ValidateVolumeCapabilities` to validate the volume capabilities, or wait for the volume to be unpublished on the node. |
 | Staging target path not set | 9 FAILED_PRECONDITION | Indicates that `STAGE_UNSTAGE_VOLUME` capability is set but no `staging_target_path` was set. | Caller MUST make sure call to `NodeStageVolume` is made and returns success before retrying with valid `staging_target_path`. |
+| Volume staged or published with different volume_mount_group | 9 FAILED_PRECONDITION | Indicates that volume with specified `volume_id` was node staged or published using different `volume_mount_group` on this node and hence can not be node published. | Caller MUST make sure that all `NodePublishVolume` calls specify same value of `volume_mount_group`. |
```

Comment:
have we considered how this feature works in light of the new single_node_multi_writer functionality being proposed?

Reply:
Currently the definition of `volume_mount_group` is more strict and requires that all node-publish RPC calls use same value of `volume_mount_group`. So basically - if `T1!=T2` and `P1!=P2` (where Pn includes `volume_mount_group`) then - for single_node_multi_writer volumes, node-publish should fail. 

Having said that - we can relax the wording once https://github.com/container-storage-interface/spec/pull/476 merges or if this merges first then we can update https://github.com/container-storage-interface/spec/pull/476. The relaxed wording will be - if SP supports both `SINGLE_NODE_MULTI_WRITER` and `VOLUME_MOUNT_GROUP`  capability then it MUST allow NodePublishing same volume multiple times on same node with different `volume_mount_group`. The cifs implementations that I know already support that (you can mount same volume multiple times on the node with different gid).

The language here is however more strict - because we wanted to start with something strict and then relax the wording (an earlier version of design did allow aformentioned flow).